ocs_ci/ocs/scale_lib.py

```diff
@@ -709,3 +709,94 @@ def increase_pods_per_worker_node_count(pods_per_node=500, pods_per_core=10):
+
+    Args:
+        no_of_pvc(int): Bulk PVC count
+        access_mode (str): PVC access_mode
+        sc_name (str): SC name for pvc creation
+
+    Returns:
+         pvc_dict_list (list): List of all PVC.yaml dicts
+
+    """
+    if access_mode == 'ReadWriteMany' and 'rbd' in sc_name:
+        volume_mode_block_flag = 1
+    else:
+        volume_mode_block_flag = 0
+
+    # Construct PVC.yaml for the no_of_required_pvc count
+    # append all the pvc.yaml dict to pvc_dict_list and return the list
+    pvc_dict_list = list()
+    for i in range(no_of_pvc):
+        pvc_name = helpers.create_unique_resource_name('test', 'pvc')
+        size = f"{random.randrange(5, 105, 5)}Gi"
+        pvc_data = templating.load_yaml(constants.CSI_PVC_YAML)
+        pvc_data['metadata']['name'] = pvc_name
+        del pvc_data['metadata']['namespace']
+        pvc_data['spec']['accessModes'] = [access_mode]
+        pvc_data['spec']['storageClassName'] = sc_name
+        pvc_data['spec']['resources']['requests']['storage'] = size
+        # Check to identify RBD_RWX PVC and add VolumeMode
+        if volume_mode_block_flag:
+            pvc_data['spec']['volumeMode'] = 'Block'
+        else:
+            pvc_data['spec']['volumeMode'] = None
+        pvc_dict_list.append(pvc_data)
+
+    return pvc_dict_list
+
+
+def check_all_pvc_reached_bound_state_in_kube_job(kube_job_obj, namespace, no_of_pvc):
+    """
+    Function to check either bulk created PVCs reached Bound state using kube_job
+
+    Args:
+        kube_job_obj (obj): Kube Job Object
+        namespace (str): Namespace of PVC's created
+        no_of_pvc (int): Bulk PVC count
+
+    Returns:
+        pvc_bound_list (list): List of all PVCs which is in Bound state.
+
+    Asserts:
+        If not all PVC reached to Bound state.
+
+    """
+    # Check all the PVC reached Bound state
+    pvc_bound_list, pvc_not_bound_list = ([] for i in range(2))
+    while_iteration_count = 0
+    while True:
+        # Get kube_job obj and fetch either all PVC's are in Bound state
+        # If not bound adding those PVCs to pvc_not_bound_list
+        job_get_output = kube_job_obj.get(namespace=namespace)
+        for i in range(no_of_pvc):
+            status = job_get_output['items'][i]['status']['phase']
+            logging.info(f"pvc {job_get_output['items'][i]['metadata']['name']} status {status}")
+            if status != 'Bound':
+                pvc_not_bound_list.append(job_get_output['items'][i]['metadata']['name'])
+
+        # Check the length of pvc_not_bound_list to decide either all PVCs reached Bound state
+        # If not then wait for 30secs and re-iterate while loop
+        if len(pvc_not_bound_list):
+            time.sleep(30)
+            while_iteration_count += 1
+            # Breaking while loop after 10 Iteration i.e. after 30*10 secs of wait_time
+            # And if PVCs still not in bound state then there will be assert.
+            if while_iteration_count >= 10:
```

Comment:
should this be based on the number of volumes rather than a blanket value?

Reply:
It's not based on number of volumes, which is going to be very high if we continue to scale more than 1500 or 3000 pvcs.
Anyway the kube_job executes all the constructed yaml sequentially, so when 100th pvc yaml is getting executed there will be already some 70+ should be in Bound state, similarly when we create 1500 there would be already 1300+ pvcs in bound state, so 5mins is good time for more than 9000+ pvcs to be in Bound state. But when we execute this tests in OCP4.5 we might see problem since there is no much parallelism in CSI for creating PVCs